src/analysis/modals/ComputeModal/GcpComputeModal/GcpComputeModal.js

```diff
@@ -292,10 +292,10 @@ export const GcpComputeModalBase = ({
     const { runtime: existingRuntime, persistentDisk: existingPersistentDisk } = getExistingEnvironmentConfig();
     const { runtime: desiredRuntime, persistentDisk: desiredPersistentDisk } = getDesiredEnvironmentConfig();
     const shouldUpdatePersistentDisk = canUpdatePersistentDisk() && !_.isEqual(desiredPersistentDisk, existingPersistentDisk);
-    const shouldDeletePersistentDisk = existingPersistentDisk && !canUpdatePersistentDisk();
-    const shouldUpdateRuntime = canUpdateRuntime() && !_.isEqual(desiredRuntime, existingRuntime);
-    const shouldDeleteRuntime = existingRuntime && !canUpdateRuntime();
-    const shouldCreateRuntime = !canUpdateRuntime() && !!desiredRuntime;
+    const shouldDeletePersistentDisk = !canUpdatePersistentDisk() && !!existingPersistentDisk;
+    const shouldUpdateRuntime = canUpdateRuntime();
+    const shouldDeleteRuntime = !shouldUpdateRuntime && !!existingRuntime;
```

Comment:
What is the difference between `canUpdateRuntime` and `shouldUpdateRuntime`?

Reply:
`shouldUpdateRuntime` is whether `applyChanges` is actually going to send a runtime PATCH request (as opposed to a delete or create). `canUpdateRuntime()` checks whether the selection state is such where an update would be allowed.

Reply:
Previously `shouldUpdateRuntime` was also checking for equality between existing runtime state and desired runtime state, but that was missing the persistent disk size differene because those fields are not on the runtime state object. `canUpdateRuntime()`, however, does check for persistent disk selection changes.

Reply:
Ahhh gotcha, thanks! Would it make sense to add this comment to the code?

Reply:
This file is so needlessly complex! Sure, I'll add comments.